internal/controller/drclusterconfig_controller.go

```diff
@@ -272,12 +288,71 @@ func (r *DRClusterConfigReconciler) processCreateOrUpdate(
 		return ctrl.Result{Requeue: true}, err
 	}
 
+	if err := r.validateS3Profiles(ctx, log, drCConfig); err != nil {
+		log.Info("Reconcile error", "error", err)
+		setDRClusterConfigConfigurationProcessedCondition(&drCConfig.Status.Conditions, drCConfig.Generation,
+			err.Error(), metav1.ConditionFalse, DRClusterConfigS3Unreachable)
+
+		return ctrl.Result{Requeue: true}, err
+	}
+
 	setDRClusterConfigConfigurationProcessedCondition(&drCConfig.Status.Conditions, drCConfig.Generation,
 		"Configuration processed and validated", metav1.ConditionTrue, DRClusterConfigConditionConfigurationProcessed)
 
 	return ctrl.Result{}, nil
 }
 
+func (r *DRClusterConfigReconciler) validateS3Profiles(ctx context.Context, log logr.Logger,
+	drCConfig *ramen.DRClusterConfig,
+) error {
+	// Fetch the ramen config resource
+	_, ramenConfig, err := ConfigMapGet(ctx, r.Client)
+	if err != nil {
+		return fmt.Errorf("config map get: %w", err)
+	}
+
+	// Iterate all profiles listed in it and check for existing healthy ones
+	for profileIdx := range ramenConfig.S3StoreProfiles {
+		// for each profile, check that it has an actual secret attached to its secretRef ID
+		profile := ramenConfig.S3StoreProfiles[profileIdx]
+		secretRef := profile.S3SecretRef
+		secret := &v1.Secret{
+			ObjectMeta: metav1.ObjectMeta{Name: secretRef.Name, Namespace: secretRef.Namespace},
+		}
+
+		if err := r.Client.Get(ctx, types.NamespacedName{
+			Namespace: secret.Namespace,
+			Name:      secret.Name,
+		}, secret); err != nil {
+			if !k8serrors.IsNotFound(err) {
+				log.Info("Reconcile error", "error", err)
+
+				return fmt.Errorf("get: %w", err)
+			}
+			// If there's no secret attached to the secretRef's namespacedname -- mark profile as unhealthy
+			setDRClusterConfigS3ReachableCondition(&drCConfig.Status.Conditions, drCConfig.Generation,
+				fmt.Sprintf("Found an unhealthy S3 profile %q for which there's no secret", profile.S3ProfileName),
+				metav1.ConditionFalse, DRClusterConfigS3Unreachable)
+		}
+
+		// Profile does have a secret. Check if it has connectivity and record in status accordingly
+		if reason, err := s3ProfileValidate(ctx, r.Client, r.ObjectStoreGetter, profile.S3ProfileName, types.NamespacedName{
+			Name: drCConfig.Name, Namespace: drCConfig.Namespace,
+		}.String(), r.Log); err != nil {
+			setDRClusterConfigS3ReachableCondition(&drCConfig.Status.Conditions, drCConfig.Generation, err.Error(),
+				metav1.ConditionFalse, reason)
+		} else {
+			// Once found a healthy S3 profile -- record to status and exit
+			setDRClusterConfigS3ReachableCondition(&drCConfig.Status.Conditions, drCConfig.Generation,
+				fmt.Sprintf("S3 profile %q is healthy", profile.S3ProfileName), metav1.ConditionTrue, DRClusterConfigS3Reachable)
+
+			break
```

Comment:
There maybe other profiles that are incorrect or not reachable, we would need to loop through all and then update the condition outside the loop to True, once all S3 stores are reachable.